ci/tests/general/general_alba.py

```diff
@@ -251,45 +451,97 @@ def remove_alba_backend(alba_backend):
             raise ValueError('Unable to remove backend {0}'.format(alba_backend.name))
 
     @staticmethod
-    def upload_file(alba_backend, namespace, file_size):
+    def upload_file(alba_backend, namespace_name, file_size, object_name=None):
         """
         Upload a file into ALBA
         :param alba_backend: ALBA backend
-        :param namespace: Namespace to upload file into
+        :param namespace_name: Namespace to upload file into
         :param file_size: Size of file
+        :param object_name: Name of the object in ALBA
         :return: None
         """
+        if object_name is None:
+            object_name = namespace_name
         contents = ''.join(random.choice(chr(random.randint(32, 126))) for _ in xrange(file_size))
         temp_file_name = tempfile.mktemp()
         with open(temp_file_name, 'wb') as temp_file:
             temp_file.write(contents)
             temp_file.flush()
 
-        GeneralAlba.execute_alba_cli_action(alba_backend, 'upload', [namespace, temp_file_name, temp_file_name], False)
+        GeneralAlba.execute_alba_cli_action(alba_backend, 'upload-object', [namespace_name, temp_file_name, object_name], False)
+
+    @staticmethod
+    def list_alba_namespaces(alba_backend, **kwargs):
+        """
+        Retrieve all namespaces
+        :param alba_backend: ALBA backend
+        :return: Namespace information
+        """
+        nss = GeneralAlba.execute_alba_cli_action(alba_backend, 'list-namespaces')
+        if len(nss) > 0:
+            actual_keys = nss[0].keys()  # Only check 1st namespace for keys, assume rest is identical
+            expected_keys = ['id', 'name', 'nsm_host_id', 'preset_name', 'state']
+            assert set(actual_keys) == set(expected_keys), 'Expected keys did not match actual keys in namespace {0}'.format(nss[0])
+
+        # Apply filter
+        if kwargs.get('name') is not None:
```

Comment:
as you're using **kwargs might be interesting to filter on other values too io name only ...